src/main/java/org/killbill/billing/plugin/stripe/StripePluginProperties.java

```diff
@@ -293,6 +293,8 @@ public static Map<String, Object> toAdditionalDataMap(final Session session, @Nu
         additionalDataMap.put("success_url", session.getSuccessUrl());
         if (pk != null) {
             additionalDataMap.put("publishable_key", pk);
+            additionalDataMap.put("setup_intent_client_secret", session.getSetupIntentObject().getClientSecret());
```

Comment:
I just re-read the code: I think this is stored in the DB via `addHppRequest`. I would just merge them manually in the body of the `buildFormDescriptor` method (the `session` object is available there), when we return `PluginHostedPaymentPageFormDescriptor`.

Reply:
https://github.com/killbill/killbill-stripe-plugin/blob/fd05e6c274efd059e723ef74a2c873d5a3c82882/src/main/java/org/killbill/billing/plugin/stripe/dao/StripeDao.java#L130

The PK is null in the `addHppRequest`

...but I will move the code

Reply:
oh, I see - very fragile though 😅 New version is much better / safer 👍